File: String/colNameForNumberExcel.cpp
//https://www.geeksforgeeks.org/find-excel-column-name-given-number/
/*
n=28
while(n>0){
    r=n%26
    if r==0{
        str+='Z'
        n=n/26-1
    }
    else{
        str+='A'
        n=n/26
    }
}
*/
#include<bits/stdc++.h>
using namespace std;
int main(){
	int t;cin>>t;
	while(t--){
		long int n;cin>>n;
		string str="";
        while(n>0){
        	int r=n%26;
        	if(r==0){
        		str+='Z';
                n=n/26-1;
        	}
        	else{
        	    std::cout << "str: " << str << "  ,  r: " << r << std::endl;
        		str+=(r-1)+'A';
        		std::cout << "str: " << str << std::endl;
        		std::cout << "n before: " << n << std::endl;
        		n/=26;
        		std::cout << "n after: " << n << std::endl;
        	}
        }
        std::cout << "str: " << str << std::endl;
        reverse(str.begin(),str.end());
	    cout<<str<<endl;
    }
	return 0;
}

File: _leet/2.2-dynamicProgram/colNameForNumberExcel.py
# Python program to find Excel column name from a 
# given column number

MAX = 50

# Function to print Excel column name 
# for a given column number
def printString(n):

	# To store result (Excel column name)
	string = ["\0"] * MAX

	# To store current index in str which is result
	i = 0

	while n > 0:
		# Find remainder
		rem = n % 26

		# if remainder is 0, then a 
		# 'Z' must be there in output
		if rem == 0:
			string[i] = 'Z'
			i += 1
			n = (n // 26) - 1
		else:
			# convert number like 25 to its alpha equivalent by
			# ord('A') + 25-1 = 'Y'
			string[i] = chr((rem - 1) + ord('A'))
			i += 1
			n = n // 26
	string[i] = '\0'

	# Reverse the string and print result
	string = string[::-1]
	print("".join(string))

# Driver program to test the above Function
# printString(26)
printString(51)
printString(52)
